iron-request.html

```diff
@@ -300,6 +309,82 @@
     },
 
     /**
+     * @param {*} body The given body of the request to try and encode.
+     * @param {?string} contentType The given content type, to infer an encoding
+     *     from.
+     * @return {?string|Object} Either the encoded body as a string, if
+     *     successful, or the unaltered body object, if no encoding could be
+     *     inferred.
+     */
+    _encodeBodyObject: function(body, contentType) {
+      if (typeof body == 'string') {
+        return body;  // Already encoded.
+      }
+      switch(contentType) {
+        case('application/json'):
+          return JSON.stringify(body);
+        case('application/x-www-form-urlencoded'):
+          return this._wwwFormUrlEncode(body);
+        case(null):  // fall through
+        case(undefined):
+          // More complex objects may be natively understood by the browser,
+          // e.g. ArrayBuffer, FormData, Document.
+          // If the user wants a non-plain object to be form encoded they'll
+          // need to specify the content type manually, we shouldn't try to
+          // infer it.
```

Comment:
On reflection I agree completely. Removed inference. If it continues to be a stumbling block for people, we can always revisit.